File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s936984238_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

var score []int
var number int

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	for read(scanner) {
		slove()
	}
}

func read(scanner *bufio.Scanner) bool {
	scanner.Scan()
	number, _ = strconv.Atoi(strings.TrimSpace(scanner.Text()))
	if number == 0 {
		return false
	}
	score = make([]int, number)
	scanner.Scan()
	input := strings.Fields(scanner.Text())
	for i := 0; i < number; i++ {
		score[i], _ = strconv.Atoi(input[i])
	}
	return true
}

func slove() {
	sort.Ints(score)
	sum := 0
	if number > 2 {
		for i := 1; i < number-1; i++ {
			sum += score[i]
		}
		fmt.Println(sum / (number - 2))
	} else {
		fmt.Println(0) // Handle the case where number is less than 3
	}
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s936984238_comments_removed.py
import sys

def main():
    while read():
        slove()

def read():
    global number, score
    number = int(input())
    if number == 0:
        return False
    score = [int(input()) for _ in range(number)]
    return True

def slove():
    score.sort()
    sum = 0
    for i in range(1, number-1):
        sum += score[i]
    print(sum // (number - 2))

if __name__ == "__main__":
    main()  # End of Code
